File: 0006-sum_square_difference/main.py
#!/usr/bin/env python

def main():
    limit = 100
    sequence = list(range(1, limit + 1))
    sum_of_squares = sum(n * n for n in sequence)
    square_of_sum = sum(sequence)
    square_of_sum *= square_of_sum
    print(square_of_sum - sum_of_squares)


if __name__ == "__main__":
    main()

File: 0006-sum_square_difference/main.rs
fn main() {
    let limit: i64 = 10;
    let sequence = 1..=limit;
    let sum_of_squares: i64 = sequence
        .clone()
        .map(|n| n * n)
        .sum();
    let mut square_of_sum: i64 = sequence.sum();
    square_of_sum *= square_of_sum;
    
    println!("{}", square_of_sum - sum_of_squares);
}
